### ISSUE TITLE
Add Description Attribute to JsonSchema.Net.Generation

### ISSUE THREAD

**Issue by DanielHabenicht** [gregsdennis/json-everything#141]
**Environment**
- Nuget Package: JsonSchema.Net.Generation
- Nuget Version: 1.7.1
- OS: Windows
- .Net Target: Core 5

**Can this library make doing something simpler?**
A common use case is to add Descriptions to an Property: 

```
        [Description("The title of the event.")]
        public string Title { get; init; }
```
It would be great if `json-everything` would support this out of the box. 

**Describe alternatives you've considered**
Implement it yourself, e.g.
```
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Json.Schema;
using Json.Schema.Generation;

namespace EnoCore.Models.Schema
{
    [AttributeUsage(AttributeTargets.Property)]
    public class Description : Attribute, IAttributeHandler
    {
        public string Value { get; }

        public Description(string value)
        {
            Value = value;
        }

        void IAttributeHandler.AddConstraints(SchemaGeneratorContext context)
        {
            var attribute = context.Attributes.OfType<Description>().FirstOrDefault();
            if (attribute == null) return;

            context.Intents.Add(new DescriptionIntent(attribute.Value));
        }
    }

    public class DescriptionIntent : ISchemaKeywordIntent
    {
        public string Description { get; private set; }

        public DescriptionIntent(string description)
        {
            Description = description;
        }

        public void Apply(JsonSchemaBuilder builder) => builder.Description(this.Description);
    }
}

```

**Additional context**
I am happy to make a PR

**Comment by gregsdennis** [comment]
This is a good addition.  We should probably add

- `title`
- `default`
- `deprecated`

as well.  These and `examples` make up the [metadata keywords](https://www.github.com/json-schema-org/json-schema-spec/tree/draft-next/meta%2Fmeta-data.json).  Not sure how examples would do, though.

---

The only reason I didn't put these in at first was that they don't do anything for validation, which I figured was the primary purpose for generating a schema.

**Comment by DanielHabenicht** [comment]
ok, I will come up with a PR next weekend. 

By the way is there a way to use the maybe already specified Attributes from `System.ComponentModel.DataAnnotations` (e.g. `Required` or `[Range(minimum: 0, maximum: 2)]`)?

**Comment by gregsdennis** [comment]
`Required` is already supported, and I already have `Minimum` and `Maximum` defined.  I think these are more explicit.

Additionally, I found a [StackOverflow question](https://stackoverflow.com/a/68322628/878701) where they didn't want to use `JsonIgnore` because it interfered with using the same model for serialization.  As such, I think it's good to have explicitly defined attributes.

I think what I'm going to do is add an option to enable `Data.Annotations` support.  I can then support things like `Range` when that's turned on.

Don't worry about a PR, I've got it.  Thanks.
